src/test/java/com/ibm/cloud/cloudant/kafka/transforms/predicates/IsDesignDocumentTest.java

```diff
@@ -52,24 +49,24 @@ public void testWrongKeyType() {
 
     @Test(expected = DataException.class)
     public void testKeyNoId() {
-        isDDoc.test(wrapInRecord(RECORD_KEY_SCHEMA, Collections.singletonMap("foo", "bar")));
+        isDDoc.test(wrapInRecord(RECORD_KEY_SCHEMA, new Struct(RECORD_KEY_SCHEMA).put("foo", "bar")));
```

Comment:
I think this test isn't really providing the same "expectation" now. Before it was testing validation in the `isDDoc.test` but my feeling is that now this is throwing a `DataException` simply from the creation of the `Struct` with an invalid field against the new schema.

As such perhaps we should just simplify it to:
```suggestion
        new Struct(RECORD_KEY_SCHEMA).put("foo", "bar");
```
to avoid giving the impression that there is some additional magic happening here. It would mostly be testing the behaviour of the `Struct` class, but does at least validate the key schema a little.